libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/routeline/RouteLineComponent.kt

```diff
@@ -9,66 +9,42 @@ import com.mapbox.maps.plugin.gestures.gestures
 import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
 import com.mapbox.maps.plugin.locationcomponent.location
 import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.route.toNavigationRoutes
 import com.mapbox.navigation.core.MapboxNavigation
-import com.mapbox.navigation.core.directions.session.RoutesObserver
-import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
 import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
-import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.dropin.component.routefetch.RoutesAction
+import com.mapbox.navigation.dropin.component.routefetch.RoutesViewModel
+import com.mapbox.navigation.dropin.extensions.flowRouteProgress
+import com.mapbox.navigation.dropin.extensions.flowRoutesUpdated
 import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.findClosestRoute
-import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.setRoutes
+import com.mapbox.navigation.ui.maps.route.line.MapboxRouteLineApiExtensions.setNavigationRouteLines
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
 import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
 import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
-import com.mapbox.navigation.ui.maps.route.line.model.RouteLine
+import com.mapbox.navigation.ui.maps.route.line.model.NavigationRouteLine
 import com.mapbox.navigation.utils.internal.InternalJobControlFactory
+import com.mapbox.navigation.utils.internal.ifNonNull
 import kotlinx.coroutines.cancelChildren
+import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 @OptIn(ExperimentalPreviewMapboxNavigationAPI::class)
 internal class RouteLineComponent(
     private val mapView: MapView,
     private val options: MapboxRouteLineOptions,
+    private val routesViewModel: RoutesViewModel,
+    private val routeLineApi: MapboxRouteLineApi = MapboxRouteLineApi(options),
+    private val routeLineView: MapboxRouteLineView = MapboxRouteLineView(options)
```

Comment:
Taking `routeLineApi` and `routeLineView` via constructor injection is something we want to avoid. The idea is to make this `RouteLineComponent` available publicly. In that case, we would not want to ask users to create these many dependencies and inject it here. While it makes testing easy, we need to figure out an alternative. One proposal is [here](https://github.com/mapbox/mapbox-navigation-android/pull/5545#discussion_r823112579)

Reply:
Yeah I'm on board for that too 👍 . Sounds like a good solution for the `CameraComponent`? That way you don't have to create the NavigationCamera before hand, but you can if you want to